framework/include/variables/MooseVariableBase.h

```diff
@@ -144,6 +144,16 @@ class MooseVariableBase : public MooseObject,
    */
   virtual unsigned int numberOfDofs() const { return _dof_indices.size(); }
 
+  /**
+   * Whether or not this variable operates on an eigen kernel
+   */
+  virtual bool eigen() const { return _is_eigen; }
+
+  /**
+   * Mark this variable as an eigen var or non-eigen var
+   */
+  virtual void eigen(bool eigen) { _is_eigen = eigen; }
```

Comment:
These are new methods with a base implementation that I don't see overridden anywhere. Can we remove `virtual`?

Reply:
will do